experimental/devices/ina219/ina219.go

```diff
@@ -0,0 +1,192 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package ina219
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"sync"
+
+	"periph.io/x/periph/conn/i2c"
+	"periph.io/x/periph/conn/mmr"
+	"periph.io/x/periph/conn/physic"
+)
+
+// Opts holds the configuration options.
+//
+// Slave Address
+//
+// Depending which pins the A1, A0 pins are connected to will change the slave
+// address. Default configuration is address 0x40 (both pins to GND). For a full
+// address table see datasheet.
+type Opts struct {
+	Address       int
+	SenseResistor physic.ElectricResistance
+	MaxCurrent    physic.ElectricCurrent
+}
+
+// DefaultOpts is the recommended default options.
+var DefaultOpts = Opts{
+	Address:       0x40,
+	SenseResistor: 100 * physic.MilliOhm,
+	MaxCurrent:    3200 * physic.MilliAmpere,
+}
+
+// New opens a handle to an ina219 sensor.
+func New(bus i2c.Bus, opts *Opts) (*Dev, error) {
+
+	i2cAddress := DefaultOpts.Address
+	if opts.Address != 0 {
+		if opts.Address < 0x40 || opts.Address > 0x4f {
+			return nil, errAddressOutOfRange
+		}
+		i2cAddress = opts.Address
+	}
+
+	senseResistor := DefaultOpts.SenseResistor
+	if opts.SenseResistor != 0 {
+		if opts.SenseResistor < 1 {
+			return nil, errSenseResistorValueInvalid
+		}
+		senseResistor = opts.SenseResistor
+	}
+
+	maxCurrent := DefaultOpts.MaxCurrent
+	if opts.MaxCurrent != 0 {
+		if opts.MaxCurrent < 1 {
+			return nil, errMaxCurrentInvalid
+		}
+		maxCurrent = opts.MaxCurrent
+	}
+
+	dev := &Dev{
+		m: mmr.Dev8{
+			Conn:  &i2c.Dev{Bus: bus, Addr: uint16(i2cAddress)},
+			Order: binary.BigEndian,
+		},
+	}
+
+	if err := dev.calibrate(senseResistor, maxCurrent); err != nil {
+		return nil, err
+	}
+
+	if err := dev.m.WriteUint16(configRegister, 0x1FFF); err != nil {
+		return nil, errWritingToConfigRegister
+	}
+
+	return dev, nil
+}
+
+// Dev is a handle to the ina219 sensor.
+type Dev struct {
+	m mmr.Dev8
+
+	mu         sync.Mutex
+	currentLSB physic.ElectricCurrent
+	powerLSB   physic.Power
+}
+
+const (
+	configRegister       = 0x00
+	shuntVoltageRegister = 0x01
+	busVoltageRegister   = 0x02
+	powerRegister        = 0x03
+	currentRegister      = 0x04
+	calibrationRegister  = 0x05
+)
+
+// Sense reads the power values from the ina219 sensor.
+func (d *Dev) Sense() (PowerMonitor, error) {
+	d.mu.Lock()
+	defer d.mu.Unlock()
+
+	var pm PowerMonitor
```

Comment:
I'd recommend to change line 139 with:
```
return PowerMonitor{
 Shunt: physic.ElectricPotential(shunt) * 10 * physic.MicroVolt,
  Voltage: physic.ElectricPotential(bus>>3) * 4 * physic.MilliVolt,
  Current: physic.ElectricCurrent(current) * d.currentLSB,
}, nil
```

That would make the function a bit more dense.